### ISSUE TITLE
kubernetes-apply: Document how to give Waypoint runner access to kubectl and k8s context for GitOps flow

### ISSUE THREAD

**Issue by briancain** [hashicorp/waypoint#1407]
**Is your feature request related to a problem? Please describe.**

To enable the GitOps flow using the 0.4.0 planned plugin `kubernetes-apply`, users must give the Waypoint runner access to kubectl and a K8s context config. The docs currently don't cover how to achieve this but say they are requirements for the plugin to work.

**Describe the solution you'd like**

An extra set of docs (maybe [here](https://tip.waypointproject.io/plugins/kubernetes#requirements)) describing how to give Waypoint runner access to the requirements needed to use the `kubernetes-apply` plugin.

**Describe alternatives you've considered**

If we can make the runner automatically configured with these then maybe no docs (or less docs) would be needed.

**Comment by mitchellh** [comment]
Good call. Did you experiment with this today? I was thinking that runners should have access to service accounts and that should just work. Maybe we need to expose something in the installation process too. But docs for sure.

**Comment by briancain** [comment]
Yep @mitchellh ! @onlydole did all the hard work so he'll have more context for what he tried. 😄 That makes sense with setting up runners access to a service account. I don't think they're set up for that now....so perhaps we need to give the runner a service account with an [API token](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token)? Not sure if we'd also have to install a `kubectl` binary too on the runner but maybe between that and setting up the runner with a service account it might work without anything extra on the user side? 🤔

**Comment by mitchellh** [comment]
Oh yeah, we totally need to get `kubectl` installed into the runner since our apply plugin shells out. That's a big thing I forgot to do with that work!

**Comment by onlydole** [comment]
Thank you both! I tried a few different methods and runs on this front today and mostly ran into issues concerning: 

* AuthN/AuthZ - container registries, Kubernetes cluster access (some are covered in the docs, some are not)
* Docker in Docker / Socket - Cloud Native Buildpack issues (the docs call this out)

Either a service account (ideally) or API token option would be excellent!

**Comment by evanphx** [comment]
This is mostly that we need to package `kubectl` in our image and then document that the pod servicetoken can be used to get access to the cluster without extra configuration.

**Comment by rhabbachi** [comment]
> document that the pod servicetoken can be used to get access to the cluster without extra configuration.

I'm getting into runners deploy/config for waypoint and would love to get a peek at any draft on this (or hight level bullet points) if any.
